src/taskgraph/optimize/strategies.py

```diff
@@ -40,9 +40,10 @@ def should_replace_task(self, task, params, deadline, arg):
                     continue
 
                 return task_id
-            except KeyError:
-                # go on to the next index path
-                pass
+            except KeyError as e:
+                logger.warning(
+                    f'Couldn\'t find cached task for index path "{index_path}". Reason: {type(e)} {e}'
```

Comment:
In most cases, the `index_path` is going to be output twice (see example in my first comment). That's because a `KeyError` just contains the key as the error message. That said this line can also throw a `KeyError`:

https://github.com/taskcluster/taskgraph/blob/41a10aaf10dbe5cfe3c5cb3fc3622bcdce8c66f5/src/taskgraph/optimize/strategies.py#L31

So, I'm tempted to keep outputting both `{index_path}` and `{e}`. 

Let me know what you think!

Reply:
I would prefer to leave the exception out of the log. If it's important to have a different error message depending on where the `KeyError` comes from, then let's do two separate `try/except` blocks with different messages.

Reply:
Fixed by raising multiple `KeyError`s and catching them as one single place